packages/aws-cdk/test/api/hotswap/appsync-mapping-templates-hotswap-deployments.test.ts

```diff
@@ -0,0 +1,309 @@
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await hotswapMockSdkProvider.tryHotswapDeployment(cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).not.toBeUndefined();
+  expect(mockUpdateFunction).toHaveBeenCalledWith({
+    apiId: 'apiId',
+    dataSourceName: 'my-datasource',
+    functionId: 'functionId',
+    functionVersion: '2018-05-29',
+    name: 'my-function',
+    requestMappingTemplate: '## original request template',
+    responseMappingTemplate: '## new response template',
+  });
+});
+
+test('does not call the updateFunction() API when it receives a change that is not a mapping template difference in a Function', async () => {
+  // GIVEN
+  setup.setCurrentCfnStackTemplate({
+    Resources: {
+      AppSyncFunction: {
+        Type: 'AWS::AppSync::FunctionConfiguration',
+        Properties: {
+          RequestMappingTemplate: '## original template',
+          Name: 'my-function',
+          DataSourceName: 'my-datasource',
+        },
+        Metadata: {
+          'aws:asset:path': 'old-path',
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        AppSyncFunction: {
+          Type: 'AWS::AppSync::FunctionConfiguration',
+          Properties: {
+            RequestMappingTemplate: '## new template',
+            Name: 'my-function',
+            DataSourceName: 'new-datasource',
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await hotswapMockSdkProvider.tryHotswapDeployment(cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).toBeUndefined();
+  expect(mockUpdateFunction).not.toHaveBeenCalled();
+});
+
+test('does not call the updateFunction() API when a resource with type that is not AWS::AppSync::FunctionConfiguration but has the same properties is changed', async () => {
+  // GIVEN
+  setup.setCurrentCfnStackTemplate({
+    Resources: {
+      AppSyncFunction: {
+        Type: 'AWS::AppSync::NotAFunctionConfiguration',
+        Properties: {
+          RequestMappingTemplate: '## original template',
+          Name: 'my-function',
+          DataSourceName: 'my-datasource',
+        },
+        Metadata: {
+          'aws:asset:path': 'old-path',
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        AppSyncFunction: {
+          Type: 'AWS::AppSync::NotAFunctionConfiguration',
+          Properties: {
+            RequestMappingTemplate: '## new template',
+            Name: 'my-resolver',
+            DataSourceName: 'my-datasource',
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await hotswapMockSdkProvider.tryHotswapDeployment(cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).toBeUndefined();
+  expect(mockUpdateFunction).not.toHaveBeenCalled();
+});
```

Comment:
Surprised none of your tests use the `setup.pushStackResourceSummaries()` helper - that means the part of getting the name(s) of resources from the deployed CloudFormation Stack is not really covered by these. We should probably fix that 😉.